### ISSUE TITLE
Make library more packaging friendly

### ISSUE THREAD

**Issue by tamiko** [taskflow/taskflow#176]
I am in the process of packaging your project for Gentoo Linux (see https://github.com/gentoo/gentoo/tree/master/dev-cpp/cpp-taskflow). And it might become necessary sometime in the future to also package your project for Debian/Ubuntu. (Upstream Linux distributions usually strongly discourage bundling of external projects and we try hard to follow these best practices when packaging deal.II - this means we would like to see your project packaged for debian/ubuntu independently at some point :smiley:).

For this I currently need to work around the following (minor) issues when packaging for Gentoo:
 - It would be nice if you would provide a tarball that does not contain bundled software, for example by adding a `.gitattributes` file to your repository that sets `/3rd-party/ export-ignore` or similar.
 - Offer a CMake flag to toggle compilation of examples and tests for quicker installation (I currently simply disable most of the compilation via the following patch https://github.com/gentoo/gentoo/blob/master/dev-cpp/cpp-taskflow/files/cpp-taskflow-2.4.0-do_not_compile_examples.patch)
 - Have an explicit `OPTION` toggle for CUDA (etc.) instead of implicitly enabling/disabling support if a cuda compiler was found.
 - Install the CMake configuration in an appropriate subdirectory (I currently apply the following patch: https://github.com/gentoo/gentoo/blob/master/dev-cpp/cpp-taskflow/files/cpp-taskflow-2.4.0-fix_installation_path.patch)

**Comment by tsung-wei-huang** [comment]
@tamiko  Thanks for your items. I will work on this and make it happen in the next release. :)

**Comment by tamiko** [comment]
@tsung-wei-huang I will make a pull request with some suggested changes in the next couple of days.

**Comment by tsung-wei-huang** [comment]
Hi @tamiko I managed to add a cmake option `TF_BUILD_CUDA` (default `ON`) in the newest master branch. You may disable all CUDA code by turning it off. The CMAKE will build CUDA when `TF_BUILD_CUDA` is `ON` and `CMAKE_CUDA_COMPILER` is found.

**Comment by Levi-Armstrong** [comment]
If anyone is interested in a Debian for this package I have one under the ppa below. I may look into how best to do version release similar to clang/

https://launchpad.net/~ros-industrial/+archive/ubuntu/ppa
